registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java

```diff
@@ -170,10 +170,15 @@ public String buildJsonStringFromPacketInfo(BiometricRecord biometricRecord, Map
 				MappingJsonConstants.VALUE);
 		List<String> locationList = Arrays.asList(location.split(","));
 		List<String> locationValues = new ArrayList<>();
-		for (String locationHirerchy : locationList)
-			if (fieldMap.get(locationHirerchy) != null)
-				locationValues.add(locationHirerchy);
-
+		for (String locationHirerchy : locationList) {
+			if (fieldMap.get(locationHirerchy) != null) {
+				if (locationHirerchy.equalsIgnoreCase(MappingJsonConstants.POSTAL_CODE)) {
```

Comment:
Dont hard code postalcode here. Read registration-processor-identity.json. There you can read values for locationHierarchyForProfiling. Store all the values here. Below is the property -

"locationHierarchyForProfiling": {
         "value": "zone,postalCode"
      }